src/parser/view_transform.c

```diff
@@ -7908,7 +7908,16 @@ pt_for_update_prepare_query_internal (PARSER_CONTEXT * parser, PT_NODE * query)
 	}
       else
 	{
-	  spec->info.spec.flag = (PT_SPEC_FLAG) (spec->info.spec.flag | PT_SPEC_FLAG_FOR_UPDATE_CLAUSE);
+	  PT_NODE *entity;
+
+	  for (entity = spec->info.spec.flat_entity_list; entity; entity = entity->next)
+	    {
+	      if (sm_check_system_class_by_name (entity->info.name.original))
+		{
+		  break;
+		}
+	      spec->info.spec.flag = (PT_SPEC_FLAG) (spec->info.spec.flag | PT_SPEC_FLAG_FOR_UPDATE_CLAUSE);
+	    }
```

Comment:
spec->info.spec.flag = (PT_SPEC_FLAG) (spec->info.spec.flag | PT_SPEC_FLAG_FOR_UPDATE_CLAUSE);
 for (entity = spec->info.spec.flat_entity_list; entity; entity = entity->next)
	    {
	      if (sm_check_system_class_by_name (entity->info.name.original))
		{
                 spec->info.spec.flag = (PT_SPEC_FLAG) (spec->info.spec.flag & ~PT_SPEC_FLAG_FOR_UPDATE_CLAUSE);
		  break;
		}
	      
	    }